components/tidb_query_expr/src/impl_compare_in.rs

```diff
@@ -155,8 +155,10 @@ impl InByCompare for DateTime {}
 
 #[derive(Debug)]
 pub struct CompareInMeta<T: Eq + Hash> {
-    lookup_set: HashSet<T>,
+    lookup_map: HashMap<T, bool>,
```

Comment:
Consider the following case: there are both `-1` and `2^63-1` (which maps to `-1`). Will this expression pass this case?

Reply:
Please also fix compile errors and lints.

Reply:
new cases are added.